R/legend.R

```diff
@@ -1,47 +1,65 @@
 #' Easily remove legend(s)
 #'
-#' With no argument, will remove all legends. Provide the name(s) of 
+#' With no argument, will remove all legends. Provide the name(s) of
 #' specific aesthetic to remove only certain legends.
 #'
 #'
 #' @md
-#' @param ... additional arguments to be passed to
-#'   \code{\link[ggplot2]{element_text}}
+#' @param ... optional name(s) specific aesthetics for which to remove the 
+#' legend
+#' @param teach print longer equivalent \code{\link[ggplot2]{ggplot2}} 
+#' expression?
 #'
-#' @return either a \code{\link[ggplot2]{theme}} object or a 
-#' \code{\link[ggplot2]{guides}}object, both of which can be used in 
-#' \code{\link[ggplot2]{ggplot2}} calls 
+#' @return either a \code{\link[ggplot2]{theme}} object or a
+#' \code{\link[ggplot2]{guides}}object, both of which can be used in
+#' \code{\link[ggplot2]{ggplot2}} calls
 #' @export
-#' @import rlang
-#' @import ggplot2
-#'
+#' @author Alicia Schep
+#' 
 #' @examples
+#'
+#' library(ggplot2)
 #' 
 #' # Remove all legends
 #' ggplot(mtcars, aes(wt, mpg, colour = cyl, size = hp)) +
 #'   geom_point() + easy_remove_legend()
-#' 
+#'
 #' # remove just size legend
 #' ggplot(mtcars, aes(wt, mpg, colour = cyl, size = hp)) +
 #'   geom_point() + easy_remove_legend("size")
-#'   
+#'
 #' # can also use:
 #' ggplot(mtcars, aes(wt, mpg, colour = cyl, size = hp)) +
 #'   geom_point() + easy_remove_legend(size)
-#'   
+#'
 #' # Remove more than one
 #' ggplot(mtcars, aes(wt, mpg, colour = cyl, size = hp)) +
 #'   geom_point() + easy_remove_legend(size, color)
-easy_remove_legend <- function(...){
-
-  vars <- exprs(...)
+easy_remove_legend <- function(..., teach = FALSE) {
+  vars <- rlang::exprs(...)
   if (length(vars) == 0) {
+    if (teach) {
+      message("easy_remove_legend call can be substituted with:\n")
+      message('theme(legend.position = "none")')
+    }
     theme(legend.position = "none")
   } else {
     inputs <- lapply(vars, function(x) FALSE)
     names(inputs) <- vars
     
+    if (teach){
+      message("easy_remove_legend call can be substituted with:")
```

Comment:
I'll run this to check in a sec, but do you need a terminating `\n`?

Reply:
Disregard this one; the otherwise empty call should have the terminating `\n` removed; creates a blank space.

Reply:
yeah the first message shouldn't have \n at end; will remove it